migrator/migrations/m_170_to_m_171_create_policy_categories_and_edges/policycategorypostgresstore/store.go

```diff
@@ -109,6 +119,58 @@ func (s *storeImpl) Upsert(ctx context.Context, obj *storage.PolicyCategory) err
 	})
 }
 
+// UpsertMany saves the state of multiple objects in the storage.
+func (s *storeImpl) UpsertMany(ctx context.Context, objs []*storage.PolicyCategory) error {
+	return pgutils.Retry(func() error {
+		// Lock since copyFrom requires a delete first before being executed.  If multiple processes are updating
+		// same subset of rows, both deletes could occur before the copyFrom resulting in unique constraint
+		// violations
+		s.mutex.Lock()
+		defer s.mutex.Unlock()
+
+		if len(objs) < batchAfter {
+			return s.upsert(ctx, objs...)
+		}
+		return s.copyFrom(ctx, objs...)
+	})
+}
+
+// DeleteMany removes the objects associated to the specified IDs from the store.
```

Comment:
we don't really need the Delete routines in the migration.  May be cleaner to remove them.

Reply:
ah, that is right, it does.  carry on then.